### PR TITLE
test versioning

### PR DESCRIPTION
**Pull request by lazarusA**
test button

### CODE REVIEW

File: docs/src/composables/versions.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  {text: 'stable', link: '/',},
+  {text: 'dev', link: '/dev/',},
+  {text: 'v0.0.3', link: '/v.0.0.3/'}
+]
+
+// shared data across instances so we load only once.
+// const versions = ref([])
+
+// const dataHost = 'https://github.com/LuxDL/DocumenterVitepress.jl/blob/gh-pages/'
+// const dataUrl = `${dataHost}/versions.js`
```
Comment (asinghvi17) [inline docs/src/composables/versions.ts:12]: Can we use this instead?

Comment (asinghvi17) [reply docs/src/composables/versions.ts:12]: This way it's synced with `versions.js`for all items. See e.g. https://geo.makie.org/versions.js

File: docs/src/.vitepress/config.mts [hunk 2]
```diff
@@ -39,10 +40,12 @@ export default defineConfig({
       { text: 'Getting Started', link: '/getting_started' },
       { text: 'Markdown', link: '/markdown-examples' },
       { text: 'Code', link: '/code_example' },
-      { text: 'API', link: '/api' }
+      { text: 'API', link: '/api' },
+      { text: 'versions',
```
Comment (asinghvi17) [inline docs/src/.vitepress/config.mts:44]:
```suggestion
      { text: 'REPLACE_ME_DOCUMENTER_VITEPRESS',
```
and then we add the current version (be it dev, preview, or v*) to it?

File: docs/src/composables/versions.ts (hunk 1 above)
Comment (lazarusA) [reply docs/src/composables/versions.ts:12]: 😄 , how? the issue is getting that list automatically? maybe from the Julia side then, read, parse and include?

File: docs/src/.vitepress/config.mts (hunk 2 above)
Comment (lazarusA) [reply docs/src/.vitepress/config.mts:44]: mmm... note that the `items` part is the thing that needs to be replaced.

Comment (asinghvi17) [reply docs/src/.vitepress/config.mts:44]: I thought you had some code in the component to make items dynamic?

File: docs/src/composables/versions.ts (hunk 1 above)
Comment (asinghvi17) [reply docs/src/composables/versions.ts:12]: We could get the list by e.g.:
```js
const dataHost = 'https://luxdl.github.io/DocumenterVitepress.jl'
const dataUrl = `${dataHost}/versions.js`
```

see e.g. https://luxdl.github.io/DocumenterVitepress.jl/versions.js

but I'm not sure how the list would be constructed - or how it should be structured!

Comment (lazarusA) [reply docs/src/composables/versions.ts:12]: this the template: 
```
versions = [
  {text: 'stable', link: '/v0.0.3/',},
  {text: 'dev', link: '/dev/',},
  {text: 'v0.0.3', link: '/v0.0.3/'}
```

**Comment by asinghvi17** [comment]
Code can be run client-side by wrapping it in a client-only div as in https://vitepress.dev/reference/runtime-api#clientonly

File: docs/src/composables/versions.ts (hunk 1 above)
Comment (asinghvi17) [reply docs/src/composables/versions.ts:12]: That could work as a first pass.  My issue with it is that it creates a static version selector, meaning that people on docs of v0.4 could not access v0.6 through the selector.

If possible, the best thing to do is probably to somehow access `versions.js` in the browser and generate a selector on the client side...

**Comment by asinghvi17** [comment]
I finally figured out how Documenter does this - it injects into `<head>` a `script src="../siteinfo.js"`, so that the doc versions are loaded then and there.

https://github.com/JuliaDocs/Documenter.jl/blob/4fe9cf1237293c530633bcf0ef183b48417f23d7/src/html/HTMLWriter.jl#L999 

So at least we know what they do.  We could probably build a Vue component from this, I think.

**Comment by asinghvi17** [comment]
We could include the Documenter version JS as a local script, then deploy a version selector based on that...

**Comment by lazarusA** [comment]
new solution, WIP,  https://github.com/LuxDL/DocumenterVitepress.jl/pull/178

**Pull request closed without merging**